boards/Makefile.common

```diff
@@ -24,90 +25,98 @@ ifeq ($(CI),true)
+# `TARGET`  : The Rust target architecture the kernel is being compiled for.
 $(call check_defined, PLATFORM)
 $(call check_defined, TARGET)
 
-# If environment variable V is non-empty, be verbose
+# If environment variable V is non-empty, be verbose.
 ifneq ($(V),)
-Q=
-VERBOSE = --verbose
+  Q =
+  VERBOSE = --verbose
 else
-Q=@
-VERBOSE =
+  Q = @
+  VERBOSE =
 endif
 
+# Ask git what version of the Tock kernel we are compiling, so we can include
+# this within the binary. If Tock is not within a git repo then we fallback to
+# a set string which should be updated with every release.
 export TOCK_KERNEL_VERSION := $(shell git describe --tags --always 2> /dev/null || echo "1.4+")
 
-
-# Validate that rustup is new enough
+# Validate that rustup is new enough.
 MINIMUM_RUSTUP_VERSION := 1.11.0
-RUSTUP_VERSION := $(strip $(word 2, $(shell rustup --version)))
+RUSTUP_VERSION := $(strip $(word 2, $(shell $(RUSTUP) --version)))
 ifeq ($(shell $(MAKEFILE_COMMON_PATH)../tools/semver.sh $(RUSTUP_VERSION) \< $(MINIMUM_RUSTUP_VERSION)), true)
-  $(warning Required tool `rustup` is out-of-date.)
-  $(warning Running `rustup update` in 3 seconds (ctrl-c to cancel))
+  $(warning Required tool `$(RUSTUP)` is out-of-date.)
+  $(warning Running `$(RUSTUP) update` in 3 seconds (ctrl-c to cancel))
   $(shell sleep 3s)
-  DUMMY := $(shell rustup update)
+  DUMMY := $(shell $(RUSTUP) update)
 endif
 
-LLVM_TOOLS_INSTALLED := $(shell rustup component list | grep 'llvm-tools-preview.*(installed)' > /dev/null; echo $$?)
+# Verify that various required Rust components are installed. All of these steps
+# only have to be done once per Rust version, but will take some time when
+# compiling for the first time.
+LLVM_TOOLS_INSTALLED := $(shell $(RUSTUP) component list | grep 'llvm-tools-preview.*(installed)' > /dev/null; echo $$?)
 ifeq ($(LLVM_TOOLS_INSTALLED),1)
-  $(shell rustup component add llvm-tools-preview)
+  $(shell $(RUSTUP) component add llvm-tools-preview)
 endif
-ifneq ($(shell rustup component list | grep rust-src),rust-src (installed))
-  $(shell rustup component add rust-src)
+ifneq ($(shell $(RUSTUP) component list | grep rust-src),rust-src (installed))
+  $(shell $(RUSTUP) component add rust-src)
 endif
-ifneq ($(shell rustup target list | grep "$(TARGET) (installed)"),$(TARGET) (installed))
-  $(shell rustup target add $(TARGET))
+ifneq ($(shell $(RUSTUP) target list | grep "$(TARGET) (installed)"),$(TARGET) (installed))
+  $(shell $(RUSTUP) target add $(TARGET))
 endif
 
+# If the user is using the standard toolchain we need to get the full path.
+# rustup should take care of this for us by putting in a proxy in .cargo/bin,
+# but until that is setup we workaround it.
 ifeq ($(TOOLCHAIN),llvm)
-  # If the user is using the standard toolchain we need to get the full path.
-  # rustup should take care of this for us by putting in a proxy in .cargo/bin,
-  # but until that is setup we workaround it.
```

Comment:
This feels like a pretty useful comment, why delete it?

Reply:
Just moved it up to match the other `ifeq` blocks.